File: adventofcode/cpp/year2019/day25/day25_test.cc
#include "adventofcode/cpp/year2019/day25/day25.h"

#include <fstream>
#include <string>

#include "gtest/gtest.h"

#include "adventofcode/cpp/adventofcode.h"

TEST(Year2019Day25, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day25/actual.in");
  std::string output = "2622472";
  adventofcode::answer_t a = day25::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

File: adventofcode/java/year2016/day25/Day25Test.java
package adventofcode.java.year2016.day25;

import java.io.FileReader;
import java.io.IOException;

import org.junit.Test;
import org.junit.Assert;

public class Day25Test {
    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day25/actual.in")) {
            var output = "192";
            var result = Day25.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }
}
